src/Startup.cpp

```diff
@@ -0,0 +1,244 @@
+#include <Wt/WApplication>
+#include <Wt/WPaintedWidget>
+#include <Wt/WContainerWidget>
+#include <Wt/WPainter>
+#include <Wt/WTable>
+#include <Wt/WEnvironment>
+
+
+
+#include <Wt/WFileUpload>
+#include <Wt/WHBoxLayout>
+#include <Wt/WVBoxLayout>
+#include <Wt/WBorderLayout>
+#include <Wt/WPushButton>
+#include <Wt/WGridLayout>
+#include <Wt/WGroupBox>
+#include <Wt/WVBoxLayout>
+#include <Wt/WHBoxLayout>
+#include <Wt/WImage>
+
+#include <Wt/Http/Request>
+#include <Wt/WCheckBox>
+#include <Wt/WLineEdit>
+#include <Wt/WDialog>
+#include <Wt/WSlider>
+#include <Wt/WText>
+#include <Wt/WFileUpload>
+#include <Wt/WLink>
+
+using namespace Wt;
+
+#include <stdlib.h>
+#include <fstream>
+#include <iostream>
+
+#include "login.h"
+#include "global.h"
+
+
+#include <Wt/WLabel>
+#include <Wt/WRandom>
+#include <Wt/WSelectionBox>
+#include <boost/filesystem/operations.hpp>
+#include <boost/filesystem/fstream.hpp>
+#include <boost/filesystem.hpp>
+
+#include "Startup.h"
+
+using namespace std;
+
+using namespace boost::filesystem; 
+
+void Startup::show_files( const path & directory, vector<string>&directories, bool recurse_into_subdirs = true )
+{
+if( exists( directory ) )
+{
+directory_iterator end ;
+for( directory_iterator iter(directory) ; iter != end ; ++iter )
+{
+cout << directory << endl;
+if ( is_directory( *iter ) )
+{
+  directories.push_back(iter->path().leaf().string() );
+
+}
+
+}
+}
+}
+
+
+
+
+
+
+
+Startup::Startup(std::string wgrass_login,WContainerWidget *parent=0)
+:WContainerWidget(parent)
+{
+
+anim = WAnimation();
+
+
+
+WContainerWidget *mainContainer = new WContainerWidget(parent);
+mainContainer->resize(1000,600);
+
+Wt::WVBoxLayout *vbox = new Wt::WVBoxLayout();
+
+mainContainer->setLayout(vbox);
+
+
+
+location = new WSelectionBox(); 
+
+Wt::WImage *image = new Wt::WImage(Wt::WLink("http://grassmac.wdfiles.com/local--files/start/startup_banner.png"));
+image->resize(50,200);
+image->setWidth(5);
+vbox->addWidget(image,0.20);
+
+
+Wt::WGridLayout *grid = new Wt::WGridLayout();
+vbox->addLayout(grid);
+
+grid->addWidget(new WText("                                   "),0,0);
+
+
+grid->addWidget(new WText("GIS Data Directory    "),0,1);
+
+
+
+Wt::WSelectionBox *datadir = new Wt::WSelectionBox();
+datadir->resize(200,25);
+datadir->addItem("/home/mayank/webgrassdata");   
```

Comment:
avoid such hardcode path everywhere. use a member variable in Startup m_GrassDataBaseDirectory and use that